api.go

```diff
@@ -542,10 +559,12 @@ func NewRaft(conf *Config, fsm FSM, logs LogStore, stable StableStore, snaps Sna
 	// to be called concurrently with a blocking RPC.
 	trans.SetHeartbeatHandler(r.processHeartbeat)
 
-	// Start the background work.
-	r.goFunc(r.run)
-	r.goFunc(r.runFSM)
-	r.goFunc(r.runSnapshots)
+	if !conf.noBackgroundWork {
```

Comment:
Wow my first code review from the god himself!

I intuitively agree about negative naming but I've never heard it codified like that.
Good point about the default code path not being in a conditional. Easier to comprehend the average use case when tracing.

I'm definitely comfortable with your above edits.
Another possibility would be to yank the `r.goFunc` calls into a separate function, and check the condition in that function, since they're logically all dispatching background work. Thoughts @catsby ?

Reply:
@catsby , I added your suggestions if you wanna give it a final 👀

Reply:
👍